### ISSUE TITLE
Bug - Check failed: stream->parent()->GetConvolveAlgorithms( conv_parameters.ShouldIncludeWinogradNonfusedAlgo<T>(), &algorithms)  Aborted (core dumped)

### ISSUE THREAD

**Issue by TrsNium** [tensorflow/tensorflow#16965]
Hello ~

I got this error when use `tf.layers.conv2d`.
I use systems as following.

ubuntu 16.04
tensorflow 1.5.0
cuda 9.0
cudnn 7

and I executed my code on nvidia-docker conteiner.
(nvidia/cuda:9.0-cudnn7-runtime-ubuntu16.04)

**Comment by sagarbhathwar** [comment]
Same issue. 

System details are:
Windows 10
Python 3.6
Tensorflow 1.5.0
Cuda 9.0 and cuDNN 7.0.5

Example Code is Tensorflow's MNIST code - https://github.com/tensorflow/tensorflow/issues/16965

The debug info:
```
name: GeForce GTX 970M major: 5 minor: 2 memoryClockRate(GHz): 1.038
pciBusID: 0000:01:00.0
totalMemory: 3.00GiB freeMemory: 2.47GiB
2018-02-13 12:24:03.845670: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1195] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: GeForce GTX 970M, pci bus id: 0000:01:00.0, compute capability: 5.2)
2018-02-13 12:24:05.082021: E C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\stream_executor\cuda\cuda_blas.cc:444] failed to create cublas handle: CUBLAS_STATUS_ALLOC_FAILED
2018-02-13 12:24:05.087986: E C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\stream_executor\cuda\cuda_blas.cc:444] failed to create cublas handle: CUBLAS_STATUS_ALLOC_FAILED
2018-02-13 12:24:05.590109: E C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\stream_executor\cuda\cuda_dnn.cc:385] could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
2018-02-13 12:24:05.595021: E C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\stream_executor\cuda\cuda_dnn.cc:352] could not destroy cudnn handle: CUDNN_STATUS_BAD_PARAM
2018-02-13 12:24:05.600328: F C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\kernels\conv_ops.cc:717] Check failed: stream->parent()->GetConvolveAlgorithms( conv_parameters.ShouldIncludeWinogradNonfusedAlgo<T>(), &algorithms)
```

**Comment by sagarbhathwar** [comment]
This bug is still present with Keras program:
`
import numpy as np
import keras
from keras.models import Sequential
from keras.layers import Dense, Dropout, Flatten
from keras.layers import Conv2D, MaxPooling2D
from keras.optimizers import SGD


x_train = np.random.random((100,100,100,3))
y_train = keras.utils.to_categorical(np.random.randint(10, size=(100, 1)), num_classes=10)
x_test = np.random.random((20,100,100,3))
y_test = keras.utils.to_categorical(np.random.randint(10, size=(20,1)), num_classes=10)


model = Sequential()


model.add(Conv2D(32, (3,3), activation='relu', input_shape=(100,100,3)))
model.add(Conv2D(32, (3,3), activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Conv2D(64, (3, 3), activation='relu'))
model.add(Conv2D(64, (3, 3), activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(256, activation='relu'))
model.add(Dropout(0.5))
model.add(Dense(10, activation='softmax'))

sgd = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
model.compile(loss='categorical_crossentropy', optimizer=sgd)

model.fit(x_train, y_train, batch_size=32, epochs=10)

`
The below snipped from stackoverflow fixes it in MNIST example
```
config = tf.ConfigProto()
config.gpu_options.allow_growth = True
session = tf.Session(config=config, ...)
```

This fix from https://stackoverflow.com/questions/41117740/tensorflow-crashes-with-cublas-status-alloc-failed

What's the problem here?

**Comment by bignamehyp** [comment]
@wang-yang can you please take a look at?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jart** [comment]
The answer appears to be on stack overflow.

```py
config = tf.ConfigProto()
config.gpu_options.allow_growth = True
session = tf.Session(config=config, ...)
```

If there's a bug in TensorFlow that we can fix here, please help us learn more and I'll reopen this issue.

**Comment by TrsNium** [comment]
When I was using Tenosorflow ver 1.2.1, I can not observe this error.
I seem that some system can observe this error from ver Tensorflow ver 1.3.0. or Cuda9.0.
And this error occurs only in convolutional operation.

System detail 
nvidia drivver varsion: 384.125
gpu : tesla v100 x4
cpu : Intel(R) Xeon(R) CPU E5-2623 v3 @ 3.00GHz
tensorflow version: 1.5.0
Cuda version: 9.0
Nvidia Docker version: 2

**Comment by B-Rydalch** [comment]
I'm having the same issue, even after applying the changes posted by Jart back in April. The code I'm executing is from Udacity deep learning tutorial assignment #4. Any help would be greatly appreciated. 

System Details: 
System: Windows 10 home 64-bit, x64-based processor
Cuda: v 9.0.176
CUDNN: v 9.0 win10x64 7.3.1.2
tf-gpu: v 1.5.0 via PIP
NVIDIA: GTX 1060 6 GiB
NVIDIA DRIVER VERSION: 417.35
python v: 3.6.7

The output does not change after applying Jart's suggestion. 
Output:
~\Documents\Udacity\Deep Learning\Assignment 4 (CNN's)> python main.py
Training set (200000, 28, 28) (200000,)
Validation set (10000, 28, 28) (10000,)
Test set (10000, 28, 28) (10000,)
Training set (200000, 28, 28, 1) (200000, 10)
Validation set (10000, 28, 28, 1) (10000, 10)
Test set (10000, 28, 28, 1) (10000, 10)
2019-01-04 15:40:09.714793: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\platform\cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX AVX2
2019-01-04 15:40:10.003545: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1105] Found device 0 with properties:
name: GeForce GTX 1060 major: 6 minor: 1 memoryClockRate(GHz): 1.6705
pciBusID: 0000:01:00.0
totalMemory: 6.00GiB freeMemory: 4.97GiB
2019-01-04 15:40:10.013346: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1195] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: GeForce GTX 1060, pci bus
id: 0000:01:00.0, compute capability: 6.1)
Initialized
2019-01-04 15:40:12.584016: E C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\stream_executor\cuda\cuda_dnn.cc:378] Loaded runtime CuDNN library: 7301 (compatibility version 7300) but source was compiled with 7003 (compatibility version 7000).  If using a binary install, upgrade your CuDNN library to match.  If building from sources, make sure the library loaded at runtime matches a compatible version specified during compile configuration.
2019-01-04 15:40:12.601433: F C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\kernels\conv_ops.cc:717] Check failed: stream->parent()->GetConvolveAlgorithms( conv_parameters.ShouldIncludeWinogradNonfusedAlgo<T>(), &algorithms)

**Comment by anton-bushuiev** [comment]
For me, downgrading `cudnn` helped. See the answer on (StackOverflow)[https://stackoverflow.com/a/72885500/17572887]
